src/chainstate/stacks/index/mod.rs

```diff
@@ -61,6 +63,55 @@ impl_byte_array_newtype!(MARFValue, u8, 40);
 impl_byte_array_message_codec!(MARFValue, 40);
 pub const MARF_VALUE_ENCODED_SIZE : u32 = 40;
 
+pub trait MarfTrieId:
+PartialEq + Clone + std::fmt::Display + std::fmt::Debug +
+rusqlite::types::ToSql + rusqlite::types::FromSql + std::convert::From<[u8; 32]> +
+std::convert::From<MARFValue> + ::net::StacksMessageCodec
+{
+    fn as_bytes(&self) -> &[u8];
+    fn to_bytes(self) -> [u8; 32];
+    fn sentinel() -> Self;
+}
+
+pub const SENTINEL_ARRAY: [u8; 32] = [255u8; 32];
+
+macro_rules! impl_marf_trie_id {
+    ($thing:ident) => {
+impl MarfTrieId for $thing {
```

Comment:
Lint: can you fix the indentation here?  Our other macros don't follow this formatting.

Reply:
Sure

Reply:
Fixed in 0f8412915c8edfa9fd0172f02ae46d13c68f9f15